WooCommerce/Classes/ViewRelated/Products/Add Product/AddProductWithAI/AddProductName/AddProductNameWithAIView.swift

```diff
@@ -0,0 +1,225 @@
+import SwiftUI
+
+final class AddProductNameWithAIHostingController: UIHostingController<AddProductNameWithAIView> {
+    init(viewModel: AddProductNameWithAIViewModel) {
+        super.init(rootView: AddProductNameWithAIView(viewModel: viewModel))
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        configureTransparentNavigationBar()
+        navigationItem.leftBarButtonItem = UIBarButtonItem(title: Localization.cancel, style: .plain, target: self, action: #selector(dismissView))
+    }
+
+    @objc
+    private func dismissView() {
+        dismiss(animated: true)
+    }
+}
+
+private extension AddProductNameWithAIHostingController {
+    enum Localization {
+        static let cancel = NSLocalizedString("Cancel", comment: "Button to dismiss Add product name screen")
+    }
+}
+
+struct AddProductNameWithAIView: View {
```

Comment:
Nit: we should add comments on new types:

```suggestion
/// View for setting name for a new product with AI.
///
struct AddProductNameWithAIView: View {
```

Reply:
Added in https://github.com/woocommerce/woocommerce-ios/pull/10723/commits/b5f83db926a700be7b2ca82727af3eeba61bdf02